Release 106.0 with .NET Standard 2.0 support

alexeyzimarev opened the pull request:
Release 106.0 with .NET Standard 2.0 support. Fixes #992

alexeyzimarev commented:
The check fails since it uses the old master branch build configuration, which is incompatible with the new version of the master branch.

mjwsteenbergen commented on RestSharp/Authenticators/HttpBasicAuthenticator.cs:
```diff
@@ -44,9 +44,7 @@ public void Authenticate(IRestClient client, IRestRequest request)
-            {
-                request.AddParameter("Authorization", this.authHeader, ParameterType.HttpHeader);
-            }
+                request.AddParameter("Authorization", authHeader, ParameterType.HttpHeader);
```
Is it not a better coding convention to always have curly braces after an if?

alexeyzimarev replied:
Why is it better? It just makes the code longer.

mjwsteenbergen replied:
It forces that:
```
if(true)
     doThis()
     doThat()
```
will never be interpreted as:
```
if(true)
{
     doThis()
     doThat()
}
```

mjwsteenbergen commented on RestSharp/Authenticators/JwtAuthenticator.cs:
```diff
@@ -33,21 +33,17 @@ public class JwtAuthenticator : IAuthenticator
-            {
-                request.AddParameter("Authorization", this.authHeader, ParameterType.HttpHeader);
-            }
+                request.AddParameter("Authorization", authHeader, ParameterType.HttpHeader);
```
Is it not a better coding convention to always have curly braces after an if?

mjwsteenbergen commented on RestSharp/Authenticators/OAuth/OAuthTools.cs:
```diff
@@ -34,93 +24,85 @@ internal static class OAuthTools
+        /// <summary>
+        ///     All text parameters are UTF-8 encoded (per section 5.1).
+        /// </summary>
+        /// <seealso cref="http://www.hueniverse.com/hueniverse/2008/10/beginners-gui-1.html" />
```
This link is dead.

alexeyzimarev replied:
Sorry, not my fault. It was there and I have done nothing.

mjwsteenbergen commented on RestSharp/Authenticators/OAuth/OAuthWorkflow.cs:
```diff
@@ -250,169 +230,130 @@ public virtual OAuthWebQueryInfo BuildClientAuthAccessTokenInfo(string method, W
         {
-            if (this.RequestTokenUrl.IsNullOrBlank())
-            {
+            if (RequestTokenUrl.IsNullOrBlank())
```
Is it not a better coding convention to always have curly braces after an if? (for this entire class)

mjwsteenbergen commented on RestSharp/RestClient.cs:
```diff
@@ -390,10 +294,50 @@ public Uri BuildUri(IRestRequest request)
+        /// </summary>
+        /// <param name="contentType">MIME content type to retrieve</param>
+        /// <returns>IDeserializer instance</returns>
+        private IDeserializer GetHandler(string contentType)
```
Is/should this method be tested?

alexeyzimarev replied:
May be but this is not something I created.

mjwsteenbergen commented on RestSharp/RestRequest.cs:
```diff
@@ -21,98 +21,69 @@
         /// <param name="resource">Resource to use for this request</param>
-        public RestRequest(Uri resource) : this(resource, Method.GET) { }
+        public RestRequest(Uri resource) : this(resource, Method.GET)
+        {
```
Is this better coding convention?

mjwsteenbergen commented on RestSharp/RestRequest.cs:
```diff
@@ -368,79 +360,79 @@ public IRestRequest AddObject(object obj, params string[] includedProperties)
-                                         Type = type
-                                     });
+            return AddParameter(new Parameter
+            {
```
Why was there so much weird formatting? I see that you fixed it all over the place.

alexeyzimarev replied:
People used tabs. Tabs are evil.

mjwsteenbergen commented on build.bat:
```diff
@@ -1,119 +1,4 @@
-:failure
-
-exit /B -1
+msbuild.exe /t:restore
```
This is an improvement 😲

alexeyzimarev commented on RestSharp.IntegrationTests/AuthenticationTests.cs:
```diff
@@ -96,70 +77,22 @@ public void Can_Authenticate_With_OAuth()
             Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
         }
 
-        //[Test]
```
No, I have no idea. This was not in my focus.

alexeyzimarev replied:
I don't like commented out code. If it should be re-written - someone should do it. There is no need for commented out code.

alexeyzimarev commented on RestSharp.IntegrationTests/RestSharp.IntegrationTests.csproj:
```diff
@@ -1,121 +1,33 @@
   </ItemGroup>
   <ItemGroup>
-    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+    <None Update="Assets\Koala.jpg">
```
Yes, assets are part of the solution.

alexeyzimarev commented on RestSharp.IntegrationTests/oAuth1Tests.cs:
```diff
@@ -383,31 +330,65 @@ public void Can_Query_Vimeo()
-                "oauth_timestamp",
-                "oauth_version",
-                "oauth_signature"
+                BaseUrl = new Uri("http://api.linkedin.com/v1"),
```
Don't ask me. I have done nothing there. This is just reformatting.

alexeyzimarev commented on RestSharp.Tests/RestSharp.Tests.csproj:
```diff
@@ -1,245 +1,108 @@
   </ItemGroup>
   <ItemGroup>
-    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
+    <None Update="SampleData\4sq.txt">
```
I don't like to do things implicitly. In addition, I just added them in VS.

alexeyzimarev commented:
My answers in short:
1) Some files were re-formatted and therefore methods appear to be "added". I have not added stuff there. I added some stuff in tests only. Thefefore questions about adding tests aren't really relevant at this point since I have no added new code.
2) Using `var` when explicit type is constructed is a good practice to avoid repetition. Also, many times the type is too long. Many languages do not use explicit types and use inferred types, it does not affect readability.
3) In general, I followed the default Resharper suggestions, which in my opinion contain one of the most comprehensive naming/formatting conventions.
4) I have added ignore for one test. No other tests were changed (except of infrastructural changes), added or removed.

alexeyzimarev commented:
Just to add to my comments:
My opinion on using `var` is the same as the first answer here: https://stackoverflow.com/questions/41479/use-of-var-keyword-in-c-sharp

Using explicit type in most cases _does not_ improve readability, it just adds noise and makes code harder to read, especially if type is very long, like `IEnumerable<ConcurrentDictionary<string, SomethingVeryImportant>>`.

mjwsteenbergen commented:
In many cases `var` can increase readability, but in [this case](https://github.com/restsharp/RestSharp/pull/1025/files#diff-38891987c163ee2d8857c8fb0997e988R70) for example, I would not be able to know if `var bytes = s.GetBytes();` returns a List<Bytes>, byte,, byte[], Dictionary<T, byte> or something else.

mjwsteenbergen commented:
> Some files were re-formatted and therefore methods appear to be "added". I have not added stuff there. I added some stuff in tests only. Thefefore questions about adding tests aren't really relevant at this point since I have no added new code.

Ok, I thought you changed things. I will create issues for the deleted tests and other thing to keep record of it.

alexeyzimarev commented:
I will check to change R# rules so one-line `if`s will not be changed. May be I need to include a `.DotSettings` file to enforce conventions. We have this in MassTransit.
